docs/topics/migrations.txt

```diff
@@ -535,6 +535,45 @@ database migrations such as ``__init__()``, ``deconstruct()``, and
 which reference the field exist. For example, after squashing migrations and
 removing the old ones, you should be able to remove the field completely.
 
+.. _moving-model-between-apps:
+
+Moving a model between apps
+===========================
+
+  .. versionadded:: 5.1
+
+Django seamlessly handles the process of transferring models from one app
+to another. This involves recognizing the moved model and automatically
+generating the necessary migrations.
+
+When we are going to move a model - that is, delete a model in one app and
+recreate it in another - we want to avoid dropping and recreating its
+database table (which would lose all data). A table rename will happen
+unless the moved model in the new app defines a ``Meta.db_table`` that matches
+the old table name.
+
+To move a model from one app to another, move the model's code (the class
+definition) from the old app's ``models.py`` file to the new app's
+``models.py``, and run the :djadmin:`makemigrations` management command.
+This command will generate migrations in both apps, with the following
+operations:
+
+- First, create the new model, making it point to the already existing table. 
+  To make Django not attempt to create that table (which would fail), mark it
+  ``managed=False`` upon creation (after this operation, we momentarily have
+  two models pointing at the same table).
+- Then, to make it possible to remove the old model without deleting the
+  table, mark it ``managed=False`` as well.
+- Finally, remove the old model and mark the new one ``managed=True``
+  (these are two migrations, as each is in its own app).
```

Comment:
I tried to follow these instructions and must have misinterpreted it and had my test data deleted or I couldn't generate migrations, so I think we might want to spell this out a bit more. 

But specifically


> To move a model from one app to another, move the model's code (the class

This already sounds like the start of instructions, so let's make it visibly part of the instructions of change this e.g 

```
To move a model from one app to another:

#. ...
#. ...
```

The very first paragraph tells you to run the `makemigrations` management command but we don't explicitly say when to migrate or do further makemigrations commands or whether to restrict these commands to a particular app.

We should also add more links to reference docs here (specifically the managed and `db_table` if needed e.g. ```:attr:`Meta.db_table <django.db.models.Options.db_table>` ```)

I think we should also consider giving example names, something like:

```
To move a model `Model1` from `app1` to `app2`:

#. ...
#. ...
```

Then to give an example confusion I had
> To move a model from one app to another, move the model's code (the class
definition) from the old app's ``models.py`` file to the new app's
``models.py``, and run the :djadmin:`makemigrations` management command.

What does "move" mean here? Does move mean "duplicate" (copy and paste) and so you have the model defined in both `app1` and `app2` or does it mean "replace" (cut and paste) and so you only have the model in `app2`? 
I'm hoping have the instructions being like 1. 2. 3. will make this very clear 🤔 

Side note: I'm wondering if we should move the instructions to a 'How to' section and link them here but let's get the instructions really clear first and then decide where they belong after 👍

Reply:
> What does "move" mean here? Does move mean "duplicate" (copy and paste) and so you have the model defined in both app1 and app2 or does it mean "replace" (cut and paste) and so you only have the model in app2?

"move" actually means moving the model's code from old app to new app(cut and paste). Let me explain with an example:
Suppose you have two apps `app1` and `app2`. In `app1`, you have 2 models `ModelA` and `ModelB` and `ModelA` has a FK to `ModelB`. In `app2`, you have `ModelC`. When you run `makemigrations`, it will create initial migrations for create model.
Now suppose you want to move `ModelB` from `app1` to `app2`, for that you will first cut the code of `ModelB` from `app1` and paste it in `app2`, Now you'll have to resolve the references to `ModelB` (in this case, FK from `ModelA` ) to point to new location of `ModelB` which is `app2.ModelB`. After this you'll just have to run `makemigrations` once and django will automatically generate all the migrations that will be required to move model at database level. Thats it!


>To move a model from one app to another, move the model's code (the class
definition) from the old app's ``models.py`` file to the new app's
``models.py``, and run the :djadmin:`makemigrations` management command.

I think this line is pretty clearly written and will make sense now. ☝️ 

>This command will generate migrations in both apps, with the following
operations:
> - First, create the new model, making it point to the already existing table. 
  To make Django not attempt to create that table (which would fail), mark it
  ``managed=False`` upon creation (after this operation, we momentarily have
  two models pointing at the same table).
> - Then, to make it possible to remove the old model without deleting the
  table, mark it ``managed=False`` as well.
> - Finally, remove the old model and mark the new one ``managed=True``
  (these are two migrations, as each is in its own app).

Now these points are just to explain the migration operations that will be generated automatically and how they will actually "move" a model under the hood. You will gain a better understanding of them once you examine the migrations generated for moving a model.

Maybe its worth adding a minimal example with two apps and few models in the "how-to" section.